interop-testing/src/main/java/io/grpc/testing/integration/TestServiceImpl.java

```diff
@@ -112,10 +123,50 @@ public void unaryCall(SimpleRequest req, StreamObserver<SimpleResponse> response
       return;
     }
 
+    if (req.getOrcaPerRpc()) {
+      try {
+        echoCallMetricsFromPayload(req.getPayload());
+      } catch (InvalidProtocolBufferException ex) {
+        responseObserver.onError(ex);
+      }
+    }
+    if (req.getOrcaOob()) {
+      try {
+        echoMetricsFromPayload(req.getPayload());
+      } catch (InvalidProtocolBufferException ex) {
+        responseObserver.onError(ex);
+      }
+    }
     responseObserver.onNext(responseBuilder.build());
     responseObserver.onCompleted();
   }
 
+  private static OrcaLoadReport echoCallMetricsFromPayload(Payload payload)
+      throws InvalidProtocolBufferException {
+    OrcaLoadReport answer = OrcaLoadReport.parseFrom(payload.getBody());
+    CallMetricRecorder recorder = CallMetricRecorder.getCurrent()
+        .recordCpuUtilizationMetric(answer.getCpuUtilization())
+        .recordMemoryUtilizationMetric(answer.getMemUtilization());
+    for (Map.Entry<String, Double> entry : answer.getUtilizationMap().entrySet()) {
+      recorder.recordUtilizationMetric(entry.getKey(), entry.getValue());
+    }
+    for (Map.Entry<String, Double> entry : answer.getRequestCostMap().entrySet()) {
+      recorder.recordCallMetric(entry.getKey(), entry.getValue());
+    }
+    return answer;
+  }
+
+  private OrcaLoadReport echoMetricsFromPayload(Payload payload)
+      throws InvalidProtocolBufferException {
+    OrcaLoadReport answer = OrcaLoadReport.parseFrom(payload.getBody());
+    metricRecorder.setCpuUtilizationMetric(answer.getCpuUtilization());
```

Comment:
So CPU/Memory must always be set, or it will be cleared? That is different behavior than the map of metrics, which only overwrite if present.